src/pathfinder/opf/opf_ship.cpp

```diff
@@ -133,15 +139,16 @@ static const DiagDirection _ship_search_directions[6][4] = {
 /** Track to "direction (& 3)" mapping. */
 static const byte _pick_shiptrack_table[6] = {DIR_NE, DIR_SE, DIR_E, DIR_E, DIR_N, DIR_N};
 
-static uint FindShipTrack(const Ship *v, TileIndex tile, DiagDirection dir, TrackBits bits, TileIndex skiptile, Track *track)
+static uint FindShipTrack(const Ship *v, TileIndex tile, DiagDirection dir, TrackBits bits, TileIndex skiptile, Track *track, uint max_penalty = OPF_MAX_LENGTH, TileIndex depottile = INVALID_TILE)
```

Comment:
I strongly dislike the "depottile" parameter, a parameter that isn't a boolean and sort-of completely changed the meaning of the function is a bad smell in my opinion. And it turns out it's not even important that it's a depot, it's just an alternate destination.

The function prototype should instead be changed to always require the destination searched for as a separate parameter from the vehicle, and the return value should always be the same.